linode/ipv6range/framework_resource.go

```diff
@@ -0,0 +1,209 @@
+			"Failed to get ipv6 range when create.",
+			err.Error(),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(data.parseIPv6RangeResourceDataComputedAttrs(ctx, ipv6rangeR)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Read(
+	ctx context.Context,
+	req resource.ReadRequest,
+	resp *resource.ReadResponse,
+) {
+	var data ResourceModel
+	client := r.Meta.Client
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	ipv6range, err := client.GetIPv6Range(ctx, data.ID.ValueString())
+	if err != nil {
+		if lerr, ok := err.(*linodego.Error); ok && (lerr.Code == 404 || lerr.Code == 405) {
+			resp.Diagnostics.AddWarning(
+				"IPv6 range does not exist.",
+				fmt.Sprintf("IPv6 range \"%s\" does not exist, removing from state.", data.ID.ValueString()),
+			)
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError(
+			"Failed to get ipv6 range.",
+			err.Error(),
+		)
+		return
+	}
+
+	data.parseIPv6RangeResourceDataNonComputedAttrs(ipv6range)
+	resp.Diagnostics.Append(data.parseIPv6RangeResourceDataComputedAttrs(ctx, ipv6range)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Update(
+	ctx context.Context,
+	req resource.UpdateRequest,
+	resp *resource.UpdateResponse,
+) {
+	var plan, state ResourceModel
+	client := r.Meta.Client
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	ipv6range, err := client.GetIPv6Range(ctx, plan.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Failed to get ipv6 range when update.",
+			err.Error(),
+		)
+		return
+	}
+
+	if !state.LinodeId.Equal(plan.LinodeId) {
+		err := client.InstancesAssignIPs(ctx, linodego.LinodesAssignIPsOptions{
+			Region: ipv6range.Region,
+			Assignments: []linodego.LinodeIPAssignment{
+				{
+					LinodeID: int(plan.LinodeId.ValueInt64()),
+					Address:  fmt.Sprintf("%s/%d", ipv6range.Range, ipv6range.Prefix),
+				},
+			},
+		})
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Failed to assign ipv6 address to instance.",
+				err.Error(),
+			)
+			return
+		}
+
+		resp.Diagnostics.Append(plan.parseIPv6RangeResourceDataComputedAttrs(ctx, ipv6range)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+		resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+	}
+}
+
+func (r *Resource) Delete(
+	ctx context.Context,
+	req resource.DeleteRequest,
+	resp *resource.DeleteResponse,
+) {
+	var data ResourceModel
+	client := r.Meta.Client
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if err := client.DeleteIPv6Range(ctx, data.ID.ValueString()); err != nil {
+		if lerr, ok := err.(*linodego.Error); ok && lerr.Code == 404 {
```

Comment:
```suggestion
		if lerr, ok := err.(*linodego.Error); ok && (lerr.Code == 404 || lerr.Code == 405) {
```
IPv6 range GET API would return 405 when the resource is missing or not under this account.